packages/jest-cli/src/test_scheduler.js

```diff
@@ -138,17 +140,31 @@ class TestScheduler {
       showStatus: !runInBand,
     });
 
+    const testRunners = Object.create(null);
+    contexts.forEach(({config}) => {
+      if (!testRunners[config.runner]) {
+        // $FlowFixMe
+        testRunners[config.runner] = new (require(config.runner))(
```

Comment:
Maybe not at this point, but eventually it would be nice to guard that expression and throw a human readable error.